python-sdk/src/astro/sql/operators/run_query.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Callable, Optional
+
+from airflow.decorators.base import get_unique_task_id
+from airflow.models import BaseOperator
+from airflow.utils.context import Context
+
+from astro.databases import create_database
+
+
+class RunQueryOperator(BaseOperator):
+    """
+    Run a SQL statement and return the result of execution
+
+    :param sql_statement: sql query to execute
+    :param conn_id: Airflow connection id
+    :param handler: A Python callable handler to run on sqlalchemy legacy cursor result
+    """
+
+    template_fields = ("sql_statement", "conn_id")
+
+    def __init__(
+        self,
+        sql_statement: str,
+        conn_id: str,
+        handler: Optional[Callable] = None,
+        **kwargs
+    ):
```

Comment:
Should we have a limit on number of records? Or do we expect users to pass it "limit" in SQL.

Regardless of what we expect, we should document what might happen if they try to pull 1000s records as an extreme example

Reply:
I was assuming that the user would pass the limit in the SQL query string for now I have improved the docstring with possible issues if they try to fetch huge number of rows.

I'm ok with adding limit param in this function let me know what you think.

Reply:
@pankajastro what if we raised a warning if the user is not using a custom backend and attempts to run a query without limit?

Something along the lines of what we did for the `DataframeOperator` to avoid the user thrashing the Airflow metadata database:

https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/sql/operators/dataframe.py#L160-L163
https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/exceptions.py#L5-L15

Reply:
There are two problems that customer face when user try to fetch lots of data
- XCOM will be overloaded 
- since the main purpose of this function is to use it in the dynamic task map expand method so it can create a huge number of parallel tasks.

So, would be in favour of adding a warning if rows will be more than a certain number precisely 1024 since dynamic task map by default allow to expand these many task

Reply:
@pankajastro and I agreed we'll be introducing a parameter to limit the results outputted from `RawSQLOperator`, which also makes sense for the run_raw_sql decorator - and the `get_value_list` will be setting it to the same limit of Airflow's task template (which seems to be by default 1024, but he'll make sure we're using the same constant/config as Airflow)

Reply:
Makes sense to me 👍 Let's make sure we document it accordingly.

Reply:
After making the changes tests are failing since `RawSQLOperator` is inherited from `DecoratedOperator` which expects `python_callable`. @tatiana  any thought how I can resolve this?